src/gigs-board/entity/project/kanban-view.jsx

```diff
@@ -0,0 +1,346 @@
+
+  get_access_control_info: () =>
+    Near.view(devHubAccountId, "get_access_control_info") ?? null,
+
+  get_all_authors: () => Near.view(devHubAccountId, "get_all_authors") ?? null,
+
+  get_all_communities: () =>
+    Near.view(devHubAccountId, "get_all_communities") ?? null,
+
+  get_all_labels: () => Near.view(devHubAccountId, "get_all_labels") ?? null,
+
+  get_community: ({ handle }) =>
+    Near.view(devHubAccountId, "get_community", { handle }) ?? null,
+
+  get_post: ({ post_id }) =>
+    Near.view(devHubAccountId, "get_post", { post_id }) ?? null,
+
+  get_posts_by_author: ({ author }) =>
+    Near.view(devHubAccountId, "get_posts_by_author", { author }) ?? null,
+
+  get_posts_by_label: ({ label }) =>
+    Near.view(nearDevGovGigsContractAccountId, "get_posts_by_label", {
+      label,
+    }) ?? null,
+
+  get_root_members: () =>
+    Near.view(devHubAccountId, "get_root_members") ?? null,
+
+  useQuery: ({ name, params }) => {
+    const initialState = { data: null, error: null, isLoading: true };
+
+    const cacheState = useCache(
+      () =>
+        Near.asyncView(devHubAccountId, ["get", name].join("_"), params ?? {})
+          .then((response) => ({
+            ...initialState,
+            data: response ?? null,
+            isLoading: false,
+          }))
+          .catch((error) => ({
+            ...initialState,
+            error: props?.error ?? error,
+            isLoading: false,
+          })),
+
+      JSON.stringify({ name, params }),
+      { subscribe: true }
+    );
+
+    return cacheState === null ? initialState : cacheState;
+  },
+};
+/* END_INCLUDE: "core/adapter/dev-hub" */
+
+const configToColumns = (config) =>
+  Object.entries(config.columns).reduce((registry, [columnId, column]) => {
+    const postIds = (
+      Near.view(nearDevGovGigsContractAccountId, "get_posts_by_label", {
+        label: column.tag,
+      }) ?? []
+    ).reverse();
+
+    return {
+      ...registry,
+
+      [columnId]: {
+        ...column,
+
+        postIds:
+          tags.required.length > 0
+            ? postIds.filter(
+                (postId) =>
+                  postTagsToIdSet(tags.required).has(postId) &&
+                  !postTagsToIdSet(tags.excluded).has(postId)
+              )
+            : postIds,
+      },
+    };
+  }, {});
+
+const ProjectKanbanView = ({
+  config,
+  metadata,
+  link,
+  onConfigureClick,
+  onDeleteClick,
+  permissions,
+}) => {
+  const configuration =
+    (config ?? null) !== null
+      ? config
+      : JSON.parse(
+          DevHub.useQuery({
+            name: "project_view_config",
+            params: { id: metadata.id },
+          }).data ?? "{}"
+        );
+
+  const columns = configToColumns(configuration);
+
+  return (
+    <div className="d-flex flex-column gap-4">
+      <div
+        className={"d-flex justify-content-end gap-3 p-3 rounded-4"}
+        style={{ backgroundColor: "#181818" }}
+      >
+        {(link ?? null) !== null ? (
```

Comment:
Looks weird. `link ?? null` checks if link is null or undefined. and set it to null if so. And then you check whether it is not null. If so display something, if not return null. Let's try to simplify this code.

pseudo code:
```
if link is null or undefined
  link = null
if link is not null
  return <>...</>
else
  return null
```

Simplify:
```
if link is null or undefined => set link to null => return null
else => return <>...</>
```

Simplify:
```
link ? <>...</> : null
```

Reply:
I just applied even more straightforward `typeof link === "string" ?`. But in some cases, `(x ?? null) !== null` is a vital workaround for dealing with JS's implicit type conversion, which I wouldn't recommend relying on.